src/extensions/default/CSSPseudoSelectorHints/main.js

```diff
@@ -0,0 +1,176 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    // Load dependent modules
+    var AppInit             = brackets.getModule("utils/AppInit"),
+        CodeHintManager     = brackets.getModule("editor/CodeHintManager"),
+        TokenUtils          = brackets.getModule("utils/TokenUtils"),
+        PseudoRulesText     = require("text!PseudoSelectors.json"),
+        PseudoRules         = JSON.parse(PseudoRulesText);
+
+
+    var TOKEN_TYPE_PSEUDO_SELECTOR = 0,
+        TOKEN_TYPE_PSEUDO_ELEMENT  = 1,
+        PUNCTUATION_CHAR           = ':';
+
+    function _getPseudoContext(token, cursorText, ctx) {
+        var slicedToken,
+            contextType = -1;
+
+        // Magic code to get around CM's 'pseudo' identification logic
+        // As per CSS3 spec :
+        // -> ':' identifies pseudo slectors
+        // -> '::' identifies pseudo elements
+        // We should strictly check for single or double occurance of ':' by slicing
+        // the line text till the token start position
+
+        if (token.state.state === "pseudo") {
+            slicedToken = cursorText.substr(0, token.start + 1).slice(-3);
+        } else if (token.type === "variable-3") {
+            slicedToken = cursorText.substr(0, token.start).slice(-3);
+        }
+        
+        if (!slicedToken) {
+            //We get here when in SCSS mode and the cursor is right after ':'
+            //Test the previous token first
+            TokenUtils.movePrevToken(ctx);
+            if (ctx.token.string === PUNCTUATION_CHAR) {
+                //We are in pseudo elemnt context ('::')
+                contextType = TOKEN_TYPE_PSEUDO_ELEMENT;
+            } else {
+                contextType = TOKEN_TYPE_PSEUDO_SELECTOR;
+            }
+        } else {
+            if (slicedToken.slice(-2) === "::") {
+                contextType = TOKEN_TYPE_PSEUDO_ELEMENT;
+            } else if (slicedToken.slice(-1) === ":") {
+                contextType = TOKEN_TYPE_PSEUDO_SELECTOR;
+            }
+        }
+
+        return contextType;
+    }
+
+    /**
+     * @constructor
+     */
+    function PsudoSelectorHints() {
+    }
+
+    // As we are only going to provide :<pseudo> name hints
+    // we should claim that we don't have hints for anything else
+    PsudoSelectorHints.prototype.hasHints = function (editor, implicitChar) {
+        var pos = editor.getCursorPos(),
+            token = editor._codeMirror.getTokenAt(pos);
+
+        this.editor = editor;
+
+        // Check if we are at ':' pseudo rule or in 'variable-3' 'def' context
+        return token.state.state === "pseudo" || token.type === "variable-3" || token.string === PUNCTUATION_CHAR;
```

Comment:
nit: this could be a function determining the state. The same statement (negated) is used in `getHints` and could be reused there. Plus, it makes testing this function in isolation easier.